core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -318,8 +318,8 @@ object TopicCommand extends Logging {
         }
 
         if(tp.hasPartitions) {
-          if (topic == Topic.GROUP_METADATA_TOPIC_NAME) {
-            throw new IllegalArgumentException("The number of partitions for the offsets topic cannot be changed.")
+          if (Topic.INTERNAL_TOPICS.contains(topic)) {
+            throw new IllegalArgumentException("The number of partitions for the offsets topic or the transaction topic cannot be changed.")
```

Comment:
Can we use comma separated string of Topic.INTERNAL_TOPICS  in the error message?